include/deal.II/hp/dof_handler.h

```diff
@@ -1090,12 +1106,16 @@ namespace hp
     std::vector<unsigned int> vertex_dof_offsets;
 
     /**
-     * Array to store the information if a cell on some level has children or
-     * not. It is used by the signal slots as a persistent buffer during the
-     * refinement, i.e. from between when pre_refinement_action is called and
-     * when post_refinement_action runs.
+     * Container to temporarily store the CellID and active FE index of
+     * cells that will be refined.
      */
-    std::vector<std::unique_ptr<std::vector<bool>>> has_children;
+    std::vector<std::pair<CellId, unsigned int>> refined_cells_fe_index;
+
+    /**
+     * Container to temporarily store the CellID and active FE index of
+     * cells that will be coarsened.
+     */
+    std::vector<std::pair<CellId, unsigned int>> coarsened_cells_fe_index;
```

Comment:
Since you're going to search into this array, why not store it as a `std::map<CellId, unsigned int>` right away?

Reply:
I thought about using `std::map` here as well, but it wasn't possible with `CellId` as a `Key`, since that particular class lacks comparison mechanics.

However, this is not an issue with `cell_iterator`, which I chose to use now.